### PR TITLE
Taras/bugfix/megolm keys

### PR DESCRIPTION
**Pull request by Taraslysak**

### CODE REVIEW

File: src/matrix/encryption/account/account.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ export class EncryptionAccount implements IEncryptionAccount {
       oneTimeKeys = {...oneTimeKeys, ...oneTimeKeyPayload};
     }
     await this.publishGeneratedKEys(deviceKeys, oneTimeKeys, oneTimeKeysAmount);
+    console.log('Keys Published!');
```
Comment (daraffleur) [inline src/matrix/encryption/account/account.ts:84]: neen to delete

File: src/matrix/encryption/groupSession/groupSession.ts [hunk 2]
```diff
@@ -34,60 +34,94 @@ export class EncryptionGroupSession implements IEncryptionGroupSession {
...
-    if (!outBoundGroupSessionInfo) {
-      outboundGroup = await EncryptionGroupSession.createOutboundGroup(chat);
-      inboundSessions = {
-        [outboundGroup.sessionId]: await EncryptionGroupSession.createInboundGroup(
-          outboundGroup.sessionKey,
-          chat,
-        ),
-      };
-    } else {
+    // if (!outBoundGroupSessionInfo && outboundGroupRequired) {
```
Comment (daraffleur) [inline src/matrix/encryption/groupSession/groupSession.ts:74]: delete commented code

File: src/matrix/encryption/groupSession/groupSession.ts [hunk 3]
```diff
@@ -34,60 +34,94 @@ export class EncryptionGroupSession implements IEncryptionGroupSession {
...
-          sessionId,
-          sessionKey,
-        );
-      }
-      inboundSessions = newInboundGroups;
     }
+    const newInboundGroups: {[sessionId: string]: InboundGroupSession} = {};
+    for (const sessionId in inboundGroupSessions) {
+      const sessionKey = inboundGroupSessions[sessionId];
+      console.log('check sessionKey', sessionKey);
```
Comment (daraffleur) [inline src/matrix/encryption/groupSession/groupSession.ts:102]: need to be deleted

File: src/matrix/encryption/groupSession/groupSession.ts [hunk 4]
```diff
@@ -115,8 +149,12 @@ export class EncryptionGroupSession implements IEncryptionGroupSession {
     if (!payload.inboundGroupSessions) {
       payload.inboundGroupSessions = {};
     }
-    payload.inboundGroupSessions[inboundGroupId] = sessionKey;
+    payload.inboundGroupSessions = {
+      ...payload.inboundGroupSessions,
+      [inboundGroupId]: sessionKey,
+    };
     console.log(payload.inboundGroupSessions[inboundGroupId]);
+    console.log('payload on inbound sessions creation', payload);
```
Comment (daraffleur) [inline src/matrix/encryption/groupSession/groupSession.ts:157]: need to be deleted

File: src/matrix/encryption/groupSessionManager/groupsessionManager.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+jest.mock('../types');
+import IOlmLib from '../types';
+import {mocked} from 'ts-jest/utils';
+
+const mockIOlm = mocked(IOlmLib, true);
+
+describe('Olm Group Session Manager', () => {
+  //   beforeAll(() => {
+  //     IOlmLi;
```
Comment (daraffleur) [inline src/matrix/encryption/groupSessionManager/groupsessionManager.test.ts:16]: need to be deleted

File: src/matrix/encryption/olmSessionManager/olmSessionManager.ts [hunk 6]
```diff
@@ -140,21 +140,32 @@ export default class OlmSessionManager implements IOlmSessionManager {
...
-
-    const userDevicesMap: IActiveDeviceList = await encryptionRequests.getDeviceList(
-      foreignUsersIdx,
-    );
-    const olmEncryptionPayload = await this.generateEncryptionPayload(
-      userDevicesMap,
-      groupKeyPayload,
+    const newMembers:
+      | Realm.Results<IMemberModel & Realm.Object>
+      | (IMemberModel & Object)[] = members.filter(
```
Comment (daraffleur) [inline src/matrix/encryption/olmSessionManager/olmSessionManager.ts:145]: make Realm Object

File: src/matrix/encryption/olmSessionManager/olmSessionManager.ts [hunk 7]
```diff
@@ -140,21 +140,32 @@ export default class OlmSessionManager implements IOlmSessionManager {
...
+      (member: IMemberModel & Object) => member.encryptionKeyRequired,
```
Comment (daraffleur) [inline src/matrix/encryption/olmSessionManager/olmSessionManager.ts:146]: make Realm.Object

**Review by kipod — APPROVED** [review]

**Pull request merged**